data-prepper-plugins/aws-plugin/src/main/java/org/opensearch/dataprepper/plugins/aws/SecretValueDecoder.java

```diff
@@ -0,0 +1,13 @@
+package org.opensearch.dataprepper.plugins.aws;
+
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse;
+
+public class SecretValueDecoder {
+    public String decode(final GetSecretValueResponse getSecretValueResponse) {
+        if (getSecretValueResponse.secretString() != null) {
+            return getSecretValueResponse.secretString();
+        } else {
+            return getSecretValueResponse.secretBinary().asUtf8String();
```

Comment:
The documentation for the response syntax states that this is returned as base64 encoded data. Should we decode this data?

Reply:
+1. We need to decode this value.

`base64.decode(getSecretValueResponse.secretBinary())`

Reply:
@dlv @dinujoh 

From my own experimentation with AWS CLI, it is returned as a base64 string:
```
{ "ARN": "arn:aws:secretsmanager:us-east-1:253613578708:secret:my-test-secrets-LZpK4A", "Name": "my-test-secrets", "VersionId": "eee3e43d-b1ea-4261-88a4-e4823e4c2044", "SecretBinary": "aGVsbG8gd29ybGQ=", "VersionStages": [ "AWSCURRENT" ], "CreatedDate": "2024-01-05T17:22:17.785000-06:00" }
```
But the SDK `getSecretValueResponse.secretBinary().asUtf8String()` converts it to the actual string value ("hello world" in this example). So I am not sure if decoding is still necessary. Probably I have some misconception here?

Reply:
Can you test with binary data in secret manager ?

Reply:
Here is the snippet:
```
String secretName = "my-test-secrets";
        Region region = Region.of("us-east-1");

        // Create a Secrets Manager client
        SecretsManagerClient client = SecretsManagerClient.builder()
                .region(region)
                .build();

        PutSecretValueRequest putSecretValueRequest = PutSecretValueRequest.builder()
                .secretId(secretName)
                .secretBinary(SdkBytes.fromUtf8String("hello world"))
                .build();
        client.putSecretValue(putSecretValueRequest);

        GetSecretValueRequest getSecretValueRequest = GetSecretValueRequest.builder()
                .secretId(secretName)
                .build();

        GetSecretValueResponse getSecretValueResponse;

        try {
            getSecretValueResponse = client.getSecretValue(getSecretValueRequest);
        } catch (Exception e) {
            // For a list of exceptions thrown, see
            // https://docs.aws.amazon.com/secretsmanager/latest/apireference/API_GetSecretValue.html
            throw e;
        }

        System.out.println(getSecretValueResponse.secretBinary().asUtf8String());
```
The output is "hello world" instead of the base64 representation string. Despite the fact that cli yields the base64 representation string. So from java SDK itself, it seems not necessary to decode the value

Reply:
Here is the Java documentation:

https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/secretsmanager/model/GetSecretValueResponse.html#secretBinary()

It says it returns "The decrypted secret value". Though it also has a confusing comment about being base64 encoded after that.

However, the documentation for the [PutSecretValue](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/secretsmanager/model/PutSecretValueRequest.html#secretString()) API is clearer.

> The text to encrypt and store in the new version of the secret. 

I think the code above demonstrates that this is what we want.

Reply:
The [Java doc](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/kms/model/DecryptResult.html) explicitly stated that: 
```
Decrypted plaintext data. When you use the HTTP API or the AWS CLI, the value is Base64-encoded. Otherwise, it is not Base64-encoded.
```